zenodo/modules/records/tasks.py

```diff
@@ -46,3 +54,60 @@ def update_expired_embargos():
     indexer = RecordIndexer()
     indexer.bulk_index(record_ids)
     indexer.process_bulk_queue()
+
+
+@shared_task(ignore_result=True, rate_limit='100/m')
+def update_datacite_metadata(pid_value, object_uuid, job_id):
+    """Update DataCite metadata of a single PersistentIdentifier."""
+
+    task_details = current_cache.get('update_datacite:task_details')
+
+    if task_details is None or job_id != task_details['job_id']:
```

Comment:
`job_id` is a string and `task_details['job_id']` a `uuid.UUID`.

```python
import uuid
u = uuid.uuid4()
assert str(u) != u
```

So, you can either store string version of it in Redis or cast to `str` here.